plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -327,6 +355,62 @@ func (s *SQLServer) accRow(query Query, acc telegraf.Accumulator, row scanner) e
 	return nil
 }
 
+// gatherHealth stores info about any query errors in the healthMetrics map
+func (s *SQLServer) gatherHealth(healthMetrics map[string]*HealthMetric, serv string, queryError error, mutex *sync.Mutex) {
+	if !s.HealthMetric {
+		return
+	}
```

Comment:
How can this happen? And shouldn't this be a big fat error in case it _really_ happens?

Reply:
This is actually the only check that enables/disables gathering the health metric values. You can see that on line 261, `gatherHealth` will always be called even when the health metric is disabled. 

I chose to do it this way to improve readability and reduce indentation in the nested for loops on lines 254 - 264 in the `Gather` method. I originally had more logic within those nested for loops. But I felt it made the code too convoluted.

Let me know what you think.

Reply:
Right I confused this with `healthMetrics`. As I outlined above, we might also want to do this check in the level above so everybody directly sees that the gathering here is optional but I can live with both solutions...

Reply:
The way you suggested below works. I'll go with that.